hrabsence/CRM/HRAbsence/BAO/HRAbsencePeriod.php

```diff
@@ -134,4 +136,21 @@ public static function getAbsencePeriods($startDate = NULL, $endDate = NULL) {
 
     return $data;
   }
+
+  /**
+   * Add a new task to the Entitlement Recalculation queue to recalculate it for
+   * the given absence period
+   *
+   * @param \CRM_HRAbsence_BAO_HRAbsencePeriod $period
```

Comment:
you only need the period id here so why not just pass the ID instead of the whole object instea